hier_config/constructors.py

```diff
@@ -0,0 +1,331 @@
+from .platforms.cisco_nxos.driver import HConfigDriverCiscoNXOS
+from .platforms.cisco_nxos.view import HConfigViewCiscoNXOS
+from .platforms.cisco_xr.driver import HConfigDriverCiscoIOSXR
+from .platforms.cisco_xr.view import HConfigViewCiscoIOSXR
+from .platforms.generic.driver import HConfigDriverGeneric
+from .platforms.hp_comware5.driver import HConfigDriverHPComware5
+from .platforms.hp_procurve.driver import HConfigDriverHPProcurve
+from .platforms.hp_procurve.view import HConfigViewHPProcurve
+from .platforms.juniper_junos.driver import HConfigDriverJuniperJUNOS
+from .platforms.view_base import HConfigViewBase
+from .platforms.vyos.driver import HConfigDriverVYOS
+from .root import HConfig
+
+logger = getLogger(__name__)
+
+
+def get_hconfig_driver(platform: Platform) -> HConfigDriverBase:  # noqa: PLR0911
+    """Create base options on an OS level."""
+    match platform:
+        case Platform.ARISTA_EOS:
+            return HConfigDriverAristaEOS()
+        case Platform.CISCO_IOS:
+            return HConfigDriverCiscoIOS()
+        case Platform.CISCO_NXOS:
+            return HConfigDriverCiscoNXOS()
+        case Platform.CISCO_XR:
+            return HConfigDriverCiscoIOSXR()
+        case Platform.GENERIC:
+            return HConfigDriverGeneric()
+        case Platform.HP_PROCURVE:
+            return HConfigDriverHPProcurve()
+        case Platform.HP_COMWARE5:
+            return HConfigDriverHPComware5()
+        case Platform.JUNIPER_JUNOS:
+            return HConfigDriverJuniperJUNOS()
+        case Platform.VYOS:
+            return HConfigDriverVYOS()
+
+
+def get_hconfig_view(config: HConfig) -> HConfigViewBase:
+    match config.driver.platform:
+        case Platform.ARISTA_EOS:
+            return HConfigViewAristaEOS(config)
+        case Platform.CISCO_IOS:
+            return HConfigViewCiscoIOS(config)
+        case Platform.CISCO_NXOS:
+            return HConfigViewCiscoNXOS(config)
+        case Platform.CISCO_XR:
+            return HConfigViewCiscoIOSXR(config)
+        case Platform.HP_PROCURVE:
+            return HConfigViewHPProcurve(config)
+        case _:
+            message = f"Unsupported platform: {config.driver.platform}"
+            raise ValueError(message)
+
+
+def get_hconfig(
+    platform_or_driver: Platform | HConfigDriverBase,
+    config_raw: Path | str = "",
+) -> HConfig:
+    if isinstance(config_raw, Path):
+        config_raw = config_raw.read_text(encoding="utf8")
+
+    config = HConfig(_get_driver(platform_or_driver))
+    for rule in config.driver.full_text_sub_rules:
+        config_raw = sub(rule.search, rule.replace, config_raw)
+
+    _load_from_string_lines(config, config_raw)
+
+    for child in tuple(config.all_children()):
+        child.delete_sectional_exit()
+
+    for callback in config.driver.post_load_callbacks:
+        callback(config)
+
+    return config
+
+
+def get_hconfig_from_dump(dump: Dump) -> HConfig:
```

Comment:
Add an arg for to override the platform/driver. For cases where the user wants to use a Generic or custom driver.